src/iexamine.cpp

```diff
@@ -810,6 +810,40 @@ void iexamine::portable_structure(player &p, const tripoint &examp)
         dropped = "null";
     }
 
+    auto check_tent_intact = [&]() -> bool {
+        int radius = dropped == "large_tent_kit" ? 2 : 1;
+        furn_id floor =
+            dropped == "tent_kit" ? f_groundsheet : f_large_groundsheet;
+        furn_id wall =
+            dropped == "tent_kit" ? f_canvas_wall : f_large_canvas_wall;
+        furn_id door =
+            dropped == "tent_kit" ? f_canvas_door : f_large_canvas_door;
+        furn_id door_opened =
+            dropped == "tent_kit" ? f_canvas_door_o : f_large_canvas_door_o;
+        furn_id center_floor =
+            dropped == "large_tent_kit" ? f_center_groundsheet : floor;
+        for( int i = -radius; i <= radius; i++ ) {
+            for( int j = -radius; j <= radius; j++ ) {
+                if( i != -radius && i != radius && j != -radius && j != radius ) {
+                    if( g->m.furn( examp.x + i, examp.y + j ) != floor && g->m.furn( examp.x + i, examp.y + j ) != center_floor ) {
```

Comment:
Seems to be accessing the same tile, could use a local var (`const furn_id&`?..). Same for the "walls" check below.

A very terse block, could use a few comments for readability. "Loop through all tiles in radius" for the `for`s above, "Check floor tiles" and "Check wall tiles" for the `if-else`... Miight not be necessary, though, if local vars are used.